library/trans.c

```diff
@@ -282,52 +282,36 @@ void dnet_trans_destroy(struct dnet_trans *t)
 		localtime_r((time_t *)&t->start.tv_sec, &tm);
 		strftime(str, sizeof(str), "%F %R:%S", &tm);
 
-		if ((t->command == DNET_CMD_READ) && (t->alloc_size >= sizeof(struct dnet_cmd) + sizeof(struct dnet_io_attr))) {
+		if ((t->command == DNET_CMD_READ || t->command == DNET_CMD_WRITE) &&
+		    (t->alloc_size >= sizeof(struct dnet_cmd) + sizeof(struct dnet_io_attr))) {
```

Comment:
This should not be evaluated for write command, which doesn't have `dnet_io_attr` in the reply